parlai/ops/ngram_repeat_block.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Wrapper for ngram_repeat_block cuda extension.
+"""
+import torch
+from torch import nn
+
+import os
+from torch.utils.cpp_extension import load
+
+current = os.getcwd()
+abspath = os.path.abspath(__file__)
+dname = os.path.dirname(abspath)
+os.chdir(dname)
+
+# import ngram_repeat_block_cuda
+
+ngram_repeat_block_cuda = load(
+    name='ngram_repeat_block_cuda',
+    sources=[
+        '../clib/cuda/ngram_repeat_block_cuda.cpp',
+        '../clib/cuda/ngram_repeat_block_cuda_kernel.cu',
+    ],
+)
+os.chdir(current)
+
+
+class NGramRepeatBlock(nn.Module):
+    """
+    Wrapper class for calling ngram_repeat_block cuda extension.
+    """
+
+    def __init__(self):
+        super(NGramRepeatBlock, self).__init__()
+
+    def forward(
+        self,
+        hypothesis,
+        context,
+        lprobs,
+        bsz,
+        step,
+        beam_size,
+        no_repeat_ngram_size,
+        if_context_blocking=False,
```

Comment:
is this parameter necessary? I.e., can't we assume that if `context` is passed, we block on it? or does this make the downstream logic easier to handle in the kernel